apps/ichat-gw/Server.cxx

```diff
@@ -751,7 +751,7 @@ Server::checkSubscription(const std::string& to, const std::string& from)
    msg.addArg("sendSubscriptionResponse");
    msg.addArg(from.c_str());
    msg.addArg(to.c_str());   
-   msg.addArg(Data((unsigned long)success).c_str());   
```

Comment:
In apps/ichat-gw/ichat-gw.cxx
<https://github.com/resiprocate/resiprocate/pull/57#discussion_r75594889>:
Yes. That too created an ambiguity in function call.




On Sun, Aug 21, 2016 at 1:56 PM, Daniel Pocock <notifications@github.com>
wrote:

> In apps/ichat-gw/Server.cxx
> <https://github.com/resiprocate/resiprocate/pull/57#discussion_r75595585>:
>
> > @@ -751,7 +751,7 @@ Server::checkSubscription(const std::string& to, const std::string& from)
> >     msg.addArg("sendSubscriptionResponse");
> >     msg.addArg(from.c_str());
> >     msg.addArg(to.c_str());
> > -   msg.addArg(Data((unsigned long)success).c_str());
>
> looking at where sendSubscriptionResponse is handled in
> JabberComponent.cxx it appears to expect a numeric type too. Some of these
> types were changed in Data.hxx constructors a while ago but the ichat-gw
> code was never updated, thanks for fixing this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/resiprocate/resiprocate/pull/57/files/d202b06b7dc3670a8a396f83f75e787cccdf231a#r75595585>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIukop87Ooiaf1UAWoGjcIPVnyc5rJM1ks5qiFkDgaJpZM4JpPdP>
> .
>



-- 


*Balram PariyarathTech Head | Excel 2016*
Computer Science and Engineering (Batch: 2013-17)
Govt. Model Engineering College, Kochi-21

[image: ☎]:  +91 8129130866
✉: balbbrl@gmail.com <p.balram96@gmail.com>